chia/cmds/cmd_classes.py

```diff
@@ -0,0 +1,322 @@
+
+            kwarg_names.append(field_name)
+            option_decorators.append(
+                click.option(
+                    *option_args["param_decls"],
+                    type=type_arg,
+                    **{k: v for k, v in option_args.items() if k not in ("param_decls", "is_command_option", "type")},
+                )
+            )
+
+    return _CommandParsingStage(
+        my_dataclass=cls,
+        my_option_decorators=option_decorators,
+        my_subclasses=subclasses,
+        my_kwarg_names=kwarg_names,
+        _needs_context=needs_context,
+    )
+
+
+def _convert_class_to_function(cls: Type[ChiaCommand]) -> SyncCmd:
+    command_parser = _generate_command_parser(cls)
+
+    if inspect.iscoroutinefunction(cls.run):
+
+        async def async_base_cmd(*args: Any, **kwargs: Any) -> None:
+            await command_parser.initialize_instance(*args, **kwargs).run()  # type: ignore[misc]
+
+        def base_cmd(*args: Any, **kwargs: Any) -> None:
+            coro = async_base_cmd(*args, **kwargs)
+            assert coro is not None
+            asyncio.run(coro)
+
+    else:
+
+        def base_cmd(*args: Any, **kwargs: Any) -> None:
+            command_parser.initialize_instance(*args, **kwargs).run()
+
+    return command_parser.apply_decorators(base_cmd)
+
+
+@dataclass_transform()
+def chia_command(cmd: click.Group, name: str, help: str) -> Callable[[Type[ChiaCommand]], Type[ChiaCommand]]:
+    def _chia_command(cls: Type[ChiaCommand]) -> Type[ChiaCommand]:
+        # The type ignores here are largely due to the fact that the class information is not preserved after being
+        # passed through the dataclass wrapper.  Not sure what to do about this right now.
+        if sys.version_info < (3, 10):  # pragma: no cover
+            # stuff below 3.10 doesn't know about kw_only
+            wrapped_cls: Type[ChiaCommand] = dataclass(  # type: ignore[assignment]
+                frozen=True,
+            )(cls)
+        else:
+            wrapped_cls: Type[ChiaCommand] = dataclass(  # type: ignore[assignment]
+                frozen=True,
+                kw_only=True,
+            )(cls)
+
+        cmd.command(name, help=help)(_convert_class_to_function(wrapped_cls))
+        return wrapped_cls
+
+    return _chia_command
+
+
+@dataclass_transform()
+def command_helper(cls: Type[Any]) -> Type[Any]:
+    if sys.version_info < (3, 10):  # stuff below 3.10 doesn't support kw_only
+        new_cls = dataclass(frozen=True)(cls)  # pragma: no cover
+    else:
+        new_cls = dataclass(frozen=True, kw_only=True)(cls)
+    setattr(new_cls, "__command_helper__", True)
+    return new_cls
+
+
+Context = Dict[str, Any]
+
+
+@dataclass(frozen=True)
+class WalletClientInfo:
+    client: WalletRpcClient
+    fingerprint: int
+    config: Dict[str, Any]
+
+
+@command_helper
+class NeedsWalletRPC:
+    context: Context = field(default_factory=dict)  # pylint: disable=invalid-field-call
```

Comment:
I see the ignore, but this does strike me as odd providing a `dict` as a default here.  What's the explanation?

Reply:
The pylint warning actually doesn't have to do with the default, it just doesn't understand about `dataclass_transform()`:

```
chia\cmds\cmd_classes.py:279:23: E3701: Invalid usage of field(), it should be used within a dataclass or the make_dataclass() function. (invalid-field-call)
```

`Context` is just an alias for `Dict[str, Any]` but it's important to be called `Context` so the parser knows you want the equivalent of `click.pass_context`.